src/mame/trs/agvision.cpp

```diff
@@ -0,0 +1,417 @@
+// license:BSD-3-Clause
+// copyright-holders: tim lindner
+/***************************************************************************
+
+    agvision.cpp
+
+    TRS-80 Radio Shack AgVision and Video Text terminals
+
+	Dynamic RAM (4, 16, or 39K) starts at $0000.
+	ROM (2K) starts at $A000 and mirrors up to $BFFF
+	Static RAM starts at $C000 for 128 bytes
+	The PIA data port a is at $FF1C
+	     control port a is at $FF1D
+		    data port b is at $FF1E
+		 control port b is at $FF1F
+
+	The Control Register is at $FF20
+	The SAM is at $FFC0
+
+	PIA:
+		Port A and B make up the keyboard matrix
+		PA7  - RS-232 recieve.
+		CA1  - The horizontal sync from the VDG.
+		CA2  - Unconnected, not verified.
+		CB1  - Modem status, carrier detect.
+		CB2  - Modem control (unverified).
+		IRQA - Connected to the 6809 FIRQ.
+		IRQB - Unconnected, not verified.
+
+	Control Register:
+		Bit 7 - RS-232 transmit.
+		Bit 6 - RS-232 DTR, or activity LED (unsure which).
+		Bit 5 - unconnected (not verified).
+		Bit 3 - VDG GM0.
+		Bit 2 - VDG GM1.
+		Bit 1 - VDG GM2.
+		Bit 0 - VDG A* / G.
+
+	Reverse engineering used with permission of George Phillips of the
+	trs80gp emulator.
+
+***************************************************************************/
+
+#include "emu.h"
+
+#include "cpu/m6809/m6809.h"
+#include "machine/6821pia.h"
+#include "machine/ram.h"
+#include "screen.h"
+#include "6883sam.h"
+#include "video/mc6847.h"
+#include "bus/rs232/rs232.h"
+
+//-------------------------------------------------
+//  MACROS / CONSTANTS
+//-------------------------------------------------
+
+#define MAINCPU_TAG     "maincpu"
+#define PIA0_TAG        "pia0"
+#define SCREEN_TAG      "screen"
+#define RS232_TAG       "rs232"
+#define SAM_TAG         "sam"
+#define VDG_TAG         "vdg"
```

Comment:
The current trend is to remove these types of legacy macros. Most of these are only used in the constructor initializer list below and the strings can be placed there anyway, and most of the other uses can be replaced by passing the relevant device finder as the argument instead.

Reply:
done.